stm32-modules/include/heater-shaker/heater-shaker/motor_task.hpp

```diff
@@ -360,9 +366,37 @@ class MotorTask {
         }
     }
 
+    // unit test this
+    // comments
+    auto set_homing_speed(
+        std::array<char, SYSTEM_WIDE_SERIAL_NUMBER_LENGTH> serial_number)
+        -> void {
+        int result{};
+        auto [ptr, ec]{std::from_chars(serial_number.begin() + 5,
```

Comment:
Oh wow I've never seen this syntax before and I love it. for others in the same boat, this is [structured binding](https://en.cppreference.com/w/cpp/language/structured_binding) e.g. destructuring, and has been in since c++17!